tree/tree/test/TChainRegressions.cxx

```diff
@@ -0,0 +1,27 @@
+#include <TChain.h>
+#include <TFile.h>
+ 
+#include "gtest/gtest.h"
+
+class TTreeCache;
+
+// ROOT-10672
+TEST(TChain, GetReadCacheBug)
+{
+   const auto treename = "tree";
+   const auto filename = "tchain_getreadcachebug.root";
+   {
+      TFile f(filename, "recreate");
```

Comment:
I am no a fan of having the test in root.git proper creating files.  I usually had to kind of these in roottest.git.

Reply:
All other TChain tests do?

```
~/W/R/r/tree (master=) git grep -i 'recreate' -- tree/test
tree/test/BulkApi.cxx:      auto hfile = TFile::Open(fFileName.c_str(), "recreate", "TTree float micro benchmark ROOT file");
tree/test/BulkApiMultiple.cxx:      auto hfile = new TFile(fFileName.c_str(), "RECREATE", "TTree float and double micro benchmark ROOT file");
tree/test/BulkApiSillyStruct.cxx:      TFile *hfile = new TFile(fFileName.c_str(), "RECREATE", "TTree silly-struct benchmark");
tree/test/BulkApiVarLength.cxx:      auto hfile = new TFile(fFileName.c_str(), "RECREATE", "TTree float micro benchmark ROOT file");
tree/test/ImplicitMT.cxx:   TFile f(ofileName, "RECREATE");
tree/test/TBranch.cxx:      TFile *file = new TFile("TBranchTestTree.root", "RECREATE");
tree/test/TChainSaveAsCxx.cxx:      TFile file1("chaintree1.root", "RECREATE");
tree/test/TChainSaveAsCxx.cxx:      TFile file2("chaintree2.root", "RECREATE");
tree/test/TChainSaveAsCxx.cxx:      TFile filefr("chaintree-fr.root", "RECREATE");
tree/test/TOffsetGeneration.cxx:      auto file = new TFile("TOffsetGeneration1.root", "RECREATE");
tree/test/TOffsetGeneration.cxx:      file = new TFile("TOffsetGeneration2.root", "RECREATE");
tree/test/TOffsetGeneration.cxx:      file = new TFile("TOffsetGeneration3.root", "RECREATE");
tree/test/TOffsetGeneration.cxx:      file = new TFile("TOffsetGeneration4.root", "RECREATE");
tree/test/TOffsetGeneration.cxx:      file = new TFile("TOffsetGeneration5.root", "RECREATE");
tree/test/TTreeClusterTest.cxx:      auto file = new TFile("TTreeClusterTest.root", "RECREATE");
tree/test/TTreeTruncatedDatatypes.cxx:   TFile f(ofileName, "RECREATE");
```

Reply:
yea well :(